src/app/Plans/components/Wizard/FilterVMsForm.tsx

```diff
@@ -130,35 +146,32 @@ const FilterVMsForm: React.FunctionComponent<IFilterVMsFormProps> = ({
         emptyStateBody={LONG_LOADING_MESSAGE}
       >
         <TreeView
-          data={filterAndConvertInventoryTree(
-            treeQuery.data || null,
-            searchText,
-            treeSelection.isItemSelected,
-            treeSelection.areAllSelected,
-            isNodeSelectable,
-            getNodeBadgeContent
-          )}
+          data={treeViewData}
           defaultAllExpanded
           hasChecks
           hasBadges
           onSearch={(event) => setSearchText(event.target.value)}
           onCheck={(_event, treeViewItem) => {
             if (treeViewItem.id === 'converted-root') {
               treeSelection.selectAll(!treeSelection.areAllSelected);
-            } else {
-              const matchingNode = findMatchingNode(treeQuery.data || null, treeViewItem.id || '');
-              const isFullyChecked = isNodeFullyChecked(
-                matchingNode,
-                treeSelection.isItemSelected,
-                isNodeSelectable
-              );
-              const nodesToSelect = findMatchingNodeAndDescendants(
-                treeQuery.data || null,
-                treeViewItem.id || '',
-                isNodeSelectable
-              );
-              if (nodesToSelect.length > 0) {
-                treeSelection.selectMultiple(nodesToSelect, !isFullyChecked);
+            } else if (treeQuery.data) {
+              const matchingPath = treeQuery.data.pathsBySelfLink[treeViewItem.id || ''];
+              if (matchingPath) {
+                const matchingNode = matchingPath[matchingPath.length - 1];
+                const isFullyChecked = isNodeFullyChecked(
```

Comment:
Not sure if this is the right line that handles this, but shouldn't the root node be indeterminate state if the children are partially selected?

<img width="487" alt="Screen Shot 2021-07-22 at 3 33 11 PM" src="https://user-images.githubusercontent.com/5942899/126699602-a6741927-cc12-4b6d-a4a4-16ddaccb911b.png">

Reply:
Yep, that's totally a bug. I think I should actually fix it in this PR before I forget how all this works. Commit coming shortly.

Reply:
Fixed and pushed.